library/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/SessionReplayPrivacyTest.kt

```diff
@@ -15,282 +17,176 @@ import android.widget.NumberPicker
-        assertThat(mappers[3].mapper).isInstanceOf(MaskAllCheckBoxMapper::class.java)
-        assertThat(mappers[4].type).isEqualTo(CheckedTextView::class.java)
-        assertThat(mappers[4].mapper).isInstanceOf(CheckedTextViewMapper::class.java)
-        assertThat(mappers[5].type).isEqualTo(Button::class.java)
-        assertThat(mappers[5].mapper).isInstanceOf(ButtonMapper::class.java)
-        assertThat(mappers[6].type).isEqualTo(EditText::class.java)
-        assertThat(mappers[6].mapper).isInstanceOf(EditTextViewMapper::class.java)
-        assertThat(mappers[7].type).isEqualTo(TextView::class.java)
-        assertThat(mappers[7].mapper).isInstanceOf(TextViewMapper::class.java)
-        assertThat(mappers[8].type).isEqualTo(ImageView::class.java)
-        assertThat(mappers[8].mapper).isInstanceOf(ViewScreenshotWireframeMapper::class.java)
-    }
-
-    @TestTargetApi(26)
-    @Test
-    fun `M return the MASK_ALL mappers W rule(MASK_ALL, above API 26)`() {
-        // When
-        val mappers = SessionReplayPrivacy.MASK_ALL.mappers()
-
-        // Then
-        assertThat(mappers.size).isEqualTo(9)
-        assertThat(mappers[0].type).isEqualTo(SeekBar::class.java)
-        assertThat(mappers[0].mapper).isInstanceOf(MaskAllSeekBarWireframeMapper::class.java)
-        assertThat(mappers[1].type).isEqualTo(SwitchCompat::class.java)
-        assertThat(mappers[1].mapper).isInstanceOf(MaskAllSwitchCompatMapper::class.java)
-        assertThat(mappers[2].type).isEqualTo(RadioButton::class.java)
-        assertThat(mappers[2].mapper).isInstanceOf(MaskAllRadioButtonMapper::class.java)
-        assertThat(mappers[3].type).isEqualTo(CheckBox::class.java)
-        assertThat(mappers[3].mapper).isInstanceOf(MaskAllCheckBoxMapper::class.java)
-        assertThat(mappers[4].type).isEqualTo(CheckedTextView::class.java)
-        assertThat(mappers[4].mapper).isInstanceOf(MaskAllCheckedTextViewMapper::class.java)
-        assertThat(mappers[5].type).isEqualTo(Button::class.java)
-        assertThat(mappers[5].mapper).isInstanceOf(ButtonMapper::class.java)
-        assertThat(mappers[6].type).isEqualTo(EditText::class.java)
-        assertThat(mappers[6].mapper).isInstanceOf(EditTextViewMapper::class.java)
-        assertThat(mappers[7].type).isEqualTo(TextView::class.java)
-        assertThat(mappers[7].mapper).isInstanceOf(MaskAllTextViewMapper::class.java)
-        assertThat(mappers[8].type).isEqualTo(ImageView::class.java)
-        assertThat(mappers[8].mapper).isInstanceOf(ViewScreenshotWireframeMapper::class.java)
-    }
+    @ParameterizedTest(name = "M return mappers W rule({1}, API Level {0}")
+    @MethodSource("provideTestArguments")
+    fun `M return mappers W rule({1}, API Level {0}`(
+        apiLevel: Int,
+        maskLevel: String,
+        expectedMappers: List<MapperTypeWrapper>
+    ) {
+        // Given
+        Build.VERSION::class.java.setStaticValue("SDK_INT", apiLevel)
```

Comment:
ideally we need to reset this value back to the original after the test (this is what `ApiLevelExtension` does), because it is a static value. Maybe it is still better to have parametrized test for below API 26 and another parametrized test for API above 26? wdyt?

Reply:
Yes, good point. I've added an afterEach hook to reset the api level